File: frontend/classes/Action.ts
interface ActionData {
  id: number
  step: number
  faction: number
  type: string
  required: boolean
  parameters: any
  completed: boolean
}

class Action {
  id: number
  step: number
  faction: number
  type: string
  required: boolean
  parameters: any
  completed: boolean

  constructor(data: ActionData) {
    this.id = data.id
    this.step = data.step
    this.faction = data.faction
    this.type = data.type
    this.required = data.required
    this.parameters = data.parameters
    this.completed = data.completed
  }
}

export default Action

File: mvp-backend/rorapp/serializers/action.py
from rest_framework import serializers
from rorapp.models import Action


# Serializer used to read actions
class ActionSerializer(serializers.ModelSerializer):
    class Meta:
        model = Action
        fields = (
            "id",
            "step",
            "faction",
            "type",
            "required",
            "parameters",
            "completed",
        )
